mmv1/third_party/terraform/tpgresource/utils.go

```diff
@@ -501,14 +515,14 @@ func CheckGoogleIamPolicy(value string) error {
 	return nil
 }
 
-// Retries an operation while the canonical error code is FAILED_PRECONDTION
+// Retries an operation while the canonical error code is FAILED_PRECONDTION or RESOURCE_EXHAUSTED
 // which indicates there is an incompatible operation already running on the
-// cluster. This error can be safely retried until the incompatible operation
+// cluster or there are the number of allowed concurrent operations running on the cluster. These errors can be safely retried until the incompatible operation
 // completes, and the newly requested operation can begin.
 func RetryWhileIncompatibleOperation(timeout time.Duration, lockKey string, f func() error) error {
 	return resource.Retry(timeout, func() *resource.RetryError {
 		if err := transport_tpg.LockedCall(lockKey, f); err != nil {
-			if IsFailedPreconditionError(err) {
+			if IsFailedPreconditionError(err) || IsQuotaError(err) {
```

Comment:
Could this function be pulled into the cluster package? It doesn't seem to be shared by any other products.

Reply:
Sure! moved to the clusters package.